ctru-rs/src/applets/swkbd.rs

```diff
@@ -497,13 +735,40 @@ impl Display for Error {
                 f,
                 "input given to the software keyboard triggered the active filters"
             ),
+            Self::ButtonPressed => write!(f, "on-screen button was pressed to exit the prompt"),
         }
     }
 }
 
 impl std::error::Error for Error {}
 
+impl From<ctru_sys::SwkbdResult> for Error {
+    fn from(value: ctru_sys::SwkbdResult) -> Self {
+        match value {
+            ctru_sys::SWKBD_INVALID_INPUT => Error::InvalidParameters,
+            ctru_sys::SWKBD_OUTOFMEM => Error::OutOfMem,
+            ctru_sys::SWKBD_HOMEPRESSED => Error::HomePressed,
+            ctru_sys::SWKBD_RESETPRESSED => Error::ResetPressed,
+            ctru_sys::SWKBD_POWERPRESSED => Error::PowerPressed,
+            ctru_sys::SWKBD_PARENTAL_OK => Error::ParentalOk,
+            ctru_sys::SWKBD_PARENTAL_FAIL => Error::ParentalFail,
+            ctru_sys::SWKBD_BANNED_INPUT => Error::BannedInput,
+            ctru_sys::SWKBD_D0_CLICK => Error::ButtonPressed,
+            ctru_sys::SWKBD_D1_CLICK0 => Error::ButtonPressed,
+            ctru_sys::SWKBD_D1_CLICK1 => Error::ButtonPressed,
+            ctru_sys::SWKBD_D2_CLICK0 => Error::ButtonPressed,
+            ctru_sys::SWKBD_D2_CLICK1 => Error::ButtonPressed,
+            ctru_sys::SWKBD_D2_CLICK2 => Error::ButtonPressed,
```

Comment:
I guess it could make sense to unify `ButtonPressed`  and the `HomePressed` (and such) variants. However, in this particular context the `SWKBD_D{n}_CLICK{n}` are referring only to the "positional" buttons put at the bottom, and the `CLICK` part is still quite misleading, since there isn't actually any form of "end" to how many times the prompt can be re-tried.

Reply:
I guess I was thinking if the user exited the keyboard via e.g. either `SWKBD_D1_CLICK0` or `SWKBD_D2_CLICK0` it might make sense to return some kind of `Left` in both cases, since the app might want to know which button was pressed to exit the keyboard. I think this only applies to buttons that are configured to submit the input, not for retries, right?

Reply:
Yeah, retries have a separate prompt with 2 fixed buttons (and which of those the user pressed can't be known). This refers to the button used to actually exit the `SoftwareKeyboard`.